src/main/java/org/openrewrite/java/logging/slf4j/InexpensiveSLF4JLoggers.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2025 the original author or authors.
+ * <p>
+ * Licensed under the Moderne Source Available License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://docs.moderne.io/licensing/moderne-source-available-license
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.logging.slf4j;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Preconditions;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.internal.ListUtils;
+import org.openrewrite.java.JavaParser;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.JavaVisitor;
+import org.openrewrite.java.MethodMatcher;
+import org.openrewrite.java.search.UsesMethod;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+
+import java.util.List;
+
+public class InexpensiveSLF4JLoggers extends Recipe {
+
+    static final MethodMatcher infoMethodMatcher = new MethodMatcher("org.slf4j.Logger debug(..)");
+    static final MethodMatcher debugMethodMatcher = new MethodMatcher("org.slf4j.Logger debug(..)");
+    static final MethodMatcher traceMethodMatcher = new MethodMatcher("org.slf4j.Logger debug(..)");
+    static final MethodMatcher errorMethodMatcher = new MethodMatcher("org.slf4j.Logger debug(..)");
+    static final MethodMatcher warnMethodMatcher = new MethodMatcher("org.slf4j.Logger debug(..)");
+    static final JavaTemplate ifEnabledThenLog = JavaTemplate
+          .builder("if(#{logger:any(org.slf4j.Logger)}.is#{}Enabled()) {}")
+          .imports("org.slf4j.Logger")
+          .javaParser(JavaParser.fromJavaVersion()
+                .classpath("slf4j-api-2.1.+"))
+          .build();
+
+    @Override
+    public String getDisplayName() {
+        return "Inexpensive SLF4J loggers";
+    }
+
+    @Override
+    public String getDescription() {
+        return ".";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return Preconditions.check(new UsesMethod<>(debugMethodMatcher), new JavaVisitor<ExecutionContext>() {
+            @Override
+            public J visitMethodInvocation(J.MethodInvocation method, ExecutionContext ctx) {
+                J.MethodInvocation m = (J.MethodInvocation) super.visitMethodInvocation(method, ctx);
+                if (debugMethodMatcher.matches(m) && !(getCursor().getParentTreeCursor().getValue() instanceof J.If)) {
```

Comment:
This `instanceof` is just a very quickly thrown together check here that ensures we do not make changes where there is already a conditional, either before the recipe runs, or after the first cycle. Can potentially be improved and pulled out as a first check, but I'll leave the details here to you.